apps/rush-lib/src/logic/pnpm/PnpmLinkManager.ts

```diff
@@ -139,22 +153,47 @@ export class PnpmLinkManager extends BaseLinkManager {
     // appropriate. This folder is usually something like:
     // C:\{uri-encoed-path-to-tgz}\node_modules\{package-name}
 
+    // e.g.:
+    //   file:projects/bentleyjs-core.tgz
+    //   file:projects/build-tools.tgz_dc21d88642e18a947127a751e00b020a
+    //   file:projects/imodel-from-geojson.tgz_request@2.88.0
+    const topLevelDependencyVersion: string | undefined =
+      pnpmShrinkwrapFile!.getTopLevelDependencyVersion(project.tempProjectName);
+
+    if (!topLevelDependencyVersion) {
+      throw new InternalError(`Cannot find top level dependency for "${project.tempProjectName}"` +
+        ` in shrinkwrap.`);
+    }
+
+    const packageId: string | undefined = pnpmShrinkwrapFile!.getPackageId(topLevelDependencyVersion);
+
+    // e.g.: projects\api-documenter.tgz
+    let relativePathToTgzFile: string;
```

Comment:
You can clean this up with a conditional:

```suggestion
    const relativePathToTgzFile: string = packageId
     ? packageId.slice('file:'.length)
     :  topLevelDependencyVersion.slice('file:'.length);
```